src/client/exercise/components/userControls/binaryChoice/BinaryChoiceAdmin.tsx

```diff
@@ -33,88 +50,142 @@ export function BinaryChoiceAdmin(bindProps: UseModelProps<UCBinaryChoice>): JSX
 
       <hr />
 
-      <h6>
-        {' '}
-        Megoldások{' '}
-        <Button
-          btn="link"
-          small
-          onAction={() => {
-            append('props.options', '')
-            append('solution', '')
-          }}
-        >
-          <PlusCircle /> Alternatív megoldás megadása
-        </Button>
-      </h6>
-      {!data.props?.options?.length && (
-        <Alert type="warning">Nincs megadva választási lehetőség</Alert>
+      <h6>Megoldások</h6>
+      {data.isDynamic && !hasName && <div>Adj nevet a megoldási mezőnek!</div>}
+      {data.isDynamic && hasName && !hasSolution && (
+        <div>
+          Definiáld a megoldásfüggvényt! Minta:
+          <CodeExample>
+            {`
+(def x [{"statement" "Ez egy igaz állítás" "solution" true},
+        {"statement" "Ez egy hamis állítás" "solution" false},
+        {"statement" "Ez egy igaz állítás egyedi címkékkel" "solution" true "true-label" "IGAZ" "false-label" "HAMIS"},
+        {"statement" "Ez egy hamis állítás egyedi címkékkel" "solution" false "true-label" "I" "false-label" "H"}])
+(def solution-${data.name} (const x))
```

Comment:
you defined the options here, not the solution
```suggestion
(def options-${data.name} (const x))
```
Also try to use descriptive variable names in the example instead of `x` you can do this:
```
(def options-map [{"statement" "Ez egy igaz állítás" "solution" true},
```
or
```
(def options-${data.name}-map [{"statement" "Ez egy igaz állítás" "solution" true},
```
you should fix it for all examples

Reply:
@maxinteger I've created separate variables for statements, options, labels, and random-order.